app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -88,7 +88,9 @@ class CheckAgencyController @Inject()
                                             (implicit authContext: AuthContext, request: Request[AnyContent]): Future[Result] = {
     agentSubscriptionConnector.getRegistration(knownFacts.utr, knownFacts.postcode) map { maybeRegistration: Option[Registration] =>
       maybeRegistration match {
-        case Some(_) => Redirect(routes.CheckAgencyController.showConfirmYourAgency())
+        case Some(Registration(Some(name))) => Redirect(routes.CheckAgencyController.showConfirmYourAgency())
+                          .flashing("agencyName" -> name, "knownFactsPostcode" -> knownFacts.postcode, "utr" -> knownFacts.utr)
+        case Some(_) => InternalServerError("No organisation name for agency found in ETMP.")
```

Comment:
What does the UI look like when this happens? Do you see the generic MDTP error page or an unstyled page with the `No organisation name for agency found in ETMP.` message? If the latter I think that we should be showing the generic MDTP error page which could be achieved by throwing an exception here.

Also I think we should be logging to Kibana when this happens (including UTR in the message) - could probably acheive this either with a Logger.warn or by throwing an exception.